catalogue_pipeline/terraform/dynamo_matcher_locktable.tf

```diff
@@ -0,0 +1,50 @@
+resource "aws_dynamodb_table" "matcher_lock_table" {
+  name           = "matcher-lock-table"
+  read_capacity  = 1
+  write_capacity = 1
+  hash_key       = "id"
+
+  attribute {
+    name = "id"
+    type = "S"
+  }
+
+  attribute {
+    name = "contextId"
+    type = "S"
+  }
+
+  global_secondary_index {
+    name            = "${var.matcher_lock_table_index}"
+    hash_key        = "contextId"
+    write_capacity  = 1
+    read_capacity   = 1
+    projection_type = "ALL"
+  }
+
+  lifecycle {
+    prevent_destroy = true
+
+    ignore_changes = [
+      "read_capacity",
+      "write_capacity",
+      "global_secondary_index", # ignore all as Terraform doesn't currently allow ignoring specifically global_secondary_index.write_capacity and global_secondary_index.read_capacity
+    ]
+  }
+}
+
+module "matcher_lock_table_dynamo_autoscaling" {
+  source = "git::https://github.com/wellcometrust/terraform.git//autoscaling/dynamodb?ref=v10.2.0"
+
+  table_name = "${aws_dynamodb_table.matcher_lock_table.name}"
+
+  enable_read_scaling     = true
+  read_target_utilization = 30
+  read_min_capacity       = 1
+  read_max_capacity       = 300
+
+  enable_write_scaling     = true
+  write_target_utilization = 30
+  write_min_capacity       = 1
+  write_max_capacity       = 300
```

Comment:
How did you decide these autoscaling limits? Naively, my guess would be that a lock table uses less capacity than a table with lots of data.

Reply:
I copied and pasted, to be honest. Anyway, the `write_target_utilization` should probably stay low 'cause we want the table to scale fast. `write_max_capacity` could be lower but I'm not sure it matters cause it will only scale as much as it needs to. Also, do failed conditional puts consume write capacity? The amount of write attempts will be more than on the graph table, it's just that we expect some of them to fail

Reply:
From [the aws documentation](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/CapacityUnitCalculations.html) 
>PutItem, UpdateItem, and DeleteItem allow conditional writes, where you specify an expression that must evaluate to true in order for the operation to succeed. If the expression evaluates to false, DynamoDB will still consume write capacity units from the table